lib/reused/option.js

```diff
@@ -0,0 +1,112 @@
+class Option {
+  static of(value) {
+    if (value instanceof Option)
+      return value;
+    if (value === null || value === undefined)
+      return none;
+    return new Some(value);
+  }
+
+  static none() {
+    return none;
+  }
+}
+
+class Some extends Option {
+  /* istanbul ignore next */
+  constructor(value) {
+    super();
+    this.value = value;
+  }
+
+  get() {
+    return this.value;
+  }
+
+  map(fn) {
+    return Option.of(fn(this.value));
+  }
+
+  filter(predicate) {
+    return predicate(this.value) ? this : none;
```

Comment:
Loose truthiness; intentional? (as opposed to `=== true`)

Reply:
Hmmm, I'm assuming that a predicate will always produce a boolean, but maybe you're right and we would benefit of a type and value check...